File: src/main/java/com/huangyueran/spark/sql/DataFrameCreate.java
package com.huangyueran.spark.sql;

import com.huangyueran.spark.utils.Constant;
import com.huangyueran.spark.utils.SparkUtils;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.DataFrameReader;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SQLContext;

/**
 * @category DataFrameCreate
 * @author huangyueran
 * @time 2019-7-24 13:58:59
 */
public class DataFrameCreate {
	public static void main(String[] args) {
		JavaSparkContext sc = SparkUtils.getLocalSparkContext(DataFrameCreate.class);

		// 创建DataFrame 读取json
		SQLContext sqlContext = new SQLContext(sc);

		DataFrameReader dataFrameReader = sqlContext.read();
		Dataset<Row> dataset = dataFrameReader.json(Constant.LOCAL_FILE_PREX +"/data/resources/people.json");

		dataset.show();

		sc.close();
	}
}

File: src/main/scala/com/hyr/spark/sql/DataFrameCreate.scala
package com.hyr.spark.sql

import com.huangyueran.spark.utils.Constant
import com.hyr.spark.utils.SparkUtils
import org.apache.spark.sql.SparkSession

/** *****************************************************************************
  * @date 2019-08-20 13:31
  * @author: <a href=mailto:huangyr>黄跃然</a>
  * @Description:
  ******************************************************************************/
object DataFrameCreate {

  def main(args: Array[String]): Unit = {
    val sparkContext = SparkUtils.getLocalSparkContext(DataFrameCreate.getClass)

    val sparkSession = SparkSession.builder().appName("DataFrameCreate").getOrCreate()
    val dataFrameReader = sparkSession.read
    val dataset = dataFrameReader.json(Constant.LOCAL_FILE_PREX + "/data/resources/people.json")
    dataset.show

    val people: Unit = dataset.createOrReplaceTempView("people")
    val frame = sparkSession.sql("show tables")
    frame.show
    val peoplesData = sparkSession sql "select * from people where age = 30"
    peoplesData.show


  }

}
